VerilogCodingStyle.md

```diff
@@ -1900,6 +1900,28 @@ separate combinational (`always_comb`) block. Ideally, sequential blocks should
 contain only a register instantiation, with perhaps a load enable or an
 increment.
 
+***All clock signals should be generated using blocking assignment even
```

Comment:
This should probably be a new section. It's currently written under `Sequential Logic (Registers)`, but it is not about registers. Perhaps `Sequential Logic (Clock Dividers)` or some such?

Reply:
I agree with you.

On my first PR it was only for clock dividers.  During the discussion on this PR @rich-ho suggested me applying this rule to clock generation in general.

> ... to add a style guide statement that all clocks should be generated using blocking assigns. I think that perhaps the wording needs to be clearer and the reason made explicit.

I agreed with him, too.  But if we do that,  we also need an example for clock generation other than clock divider, so I added it.

As you wrote this rule is not for the section "Sequential Logic (Registers)" now.

I have separated it into a section called “Generating Clock”.
This time I did not squash the commits so that the changes are visible.

How about this?